prow/scripts/cluster-integration/kyma-integration-gardener-eventing-upgrade.sh

```diff
@@ -0,0 +1,136 @@
+export TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS="${TEST_INFRA_SOURCES_DIR}/prow/scripts/cluster-integration/helpers"
+export EVENTMESH_SECRET_FILE="${CREDENTIALS_DIR}/serviceKey" # For eventing E2E fast-integration tests
+
+# shellcheck source=prow/scripts/lib/log.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/log.sh"
+# shellcheck source=prow/scripts/lib/utils.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/utils.sh"
+# shellcheck source=prow/scripts/lib/kyma.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/kyma.sh"
+# shellcheck source=prow/scripts/cluster-integration/helpers/eventing.sh
+source "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/eventing.sh"
+# shellcheck source=prow/scripts/lib/gardener/gardener.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/gardener.sh"
+
+# All provides require these values, each of them may check for additional variables
+requiredVars=(
+    BOT_GITHUB_TOKEN
+    GARDENER_PROVIDER
+    KYMA_PROJECT_DIR
+    GARDENER_REGION
+    GARDENER_ZONES
+    GARDENER_CLUSTER_VERSION
+    GARDENER_KYMA_PROW_KUBECONFIG
+    GARDENER_KYMA_PROW_PROJECT_NAME
+    GARDENER_KYMA_PROW_PROVIDER_SECRET_NAME
+    CREDENTIALS_DIR
+    EVENTMESH_SECRET_FILE
+)
+
+utils::check_required_vars "${requiredVars[@]}"
+log::info "### Starting pipeline"
+
+if [[ $GARDENER_PROVIDER == "azure" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/azure.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/azure.sh"
+elif [[ $GARDENER_PROVIDER == "aws" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/aws.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/aws.sh"
+elif [[ $GARDENER_PROVIDER == "gcp" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/gcp.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/gcp.sh"
+else
+    log::error "GARDENER_PROVIDER ${GARDENER_PROVIDER} is not yet supported"
+    exit 1
+fi
+
+# nice cleanup on exit, be it successful or on fail
+trap gardener::cleanup EXIT INT
+
+#Used to detect errors for logging purposes
+ERROR_LOGGING_GUARD="true"
+export ERROR_LOGGING_GUARD
+
+RANDOM_NAME_SUFFIX=$(LC_ALL=C tr -dc 'a-z0-9' < /dev/urandom | head -c6)
+readonly COMMON_NAME_PREFIX="grd"
+COMMON_NAME=$(echo "${COMMON_NAME_PREFIX}${RANDOM_NAME_SUFFIX}" | tr "[:upper:]" "[:lower:]")
+export COMMON_NAME
+
+### Cluster name must be less than 10 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+# set pipefail to handle right errors from tests
+set -o pipefail
+
+# Install Kyma form latest release
+kyma::get_last_release_version -t "${BOT_GITHUB_TOKEN}"
+
+export KYMA_SOURCE="${kyma_get_last_release_version_return_version:?}"
+log::info "### Reading release version from RELEASE_VERSION file, got: ${KYMA_SOURCE}"
+
+# checks required vars and initializes gcloud/docker if necessary
+gardener::init
+
+# if MACHINE_TYPE is not set then use default one
+gardener::set_machine_type
+
+kyma::install_cli
```

Comment:
Are we using the last stable CLI release? Is that enough? What if we are upgrading to our PR and some unreleased commits on main (which would be also in the PR branch) require some pre-upgrade action which is not in the last release? Would it even be possible to always upgrade, if we are not using the latest reconciler from `main`?

Reply:
I use the kyma-unstable-cli now, which uses the latest reconciler

Reply:
I am not sure that we are using unstable cli here. It doesn't look like it. `kyma::install_cli` implementation seems to use stable.

Reply:
Maybe we should discuss this in the team, in a tech talk. This implies that if someone has PR for changes on reconciler and a matching PR on eventing, there is a certain ordering involved. One has to wait at least for an unstable build to be able to upgrade to a PR Kyma version which requires the reconciler change to work/upgrade properly.

Reply:
So, this pr uses the `kyma::install_cli` which uses the latest stable cli, which does not use the latest reconciler, but this is the same in our pre-submit-test-job, the only difference from this prowjob is the upgrade step. Imo using the latest reconciler could break our test pipeline and we would need to wait for reconciler team to fix it. If we still want to have a prowjob, which uses the latest reconciler, perhaps we can have an optional job, which could be triggered on demand.

Reply:
Latest unstable CLI is built and published only when all tests are green. As I mentioned before, my problem is the case, where we have a PR on Kyma that needs changes in our eventing components reconciler. In those cases, this job will always fail. Wouldn't it?

Reply:
As discussed with Tobias, up from now we will use the unstable cli version, which uses the latest reconciler in order to be up-to-date with the reconciler changes